src/core/net/ip6.cpp

```diff
@@ -1027,6 +1030,12 @@ Error Ip6::ProcessReceiveCallback(Message &          aMessage,
     // Do not forward IPv6 packets that exceed kMinimalMtu.
     VerifyOrExit(aMessage.GetLength() <= kMinimalMtu, error = kErrorDrop);
 
+#if OPENTHREAD_CONFIG_IP6_BR_COUNTERS_ENABLE
+    header.ParseFrom(aMessage);
+    UpdateBorderRoutingCounters(aMessage, header, mBorderRoutingCounters.mInboundUnicast,
+                                mBorderRoutingCounters.mInboundMulticast);
+#endif
```

Comment:
Can add `{ }` block and move the `Header header` definition here? (then we won't need `OT_UNUSED_VARIABLE()`)
```suggestion
#if OPENTHREAD_CONFIG_IP6_BR_COUNTERS_ENABLE
    {
         Header header; 
         
         header.ParseFrom(aMessage);
         UpdateBorderRoutingCounters(aMessage, header, mBorderRoutingCounters.mInboundUnicast,
                                   mBorderRoutingCounters.mInboundMulticast);
    }
#endif
```